src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Internal/MsQuicNativeMethods.cs

```diff
@@ -0,0 +1,594 @@
+{
+    /// <summary>
+    /// Contains all native delegates and structs that are used with MsQuic.
+    /// </summary>
+    internal static unsafe class MsQuicNativeMethods
+    {
+        internal const string dllName = "msquic";
+
+#pragma warning disable BCL0015 // Disable Pinvoke analyzer errors for now
+        [DllImport(dllName)]
+        internal static extern int MsQuicOpen(int version, out NativeApi* registration);
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct NativeApi
+        {
+            internal uint Version;
+
+            internal IntPtr SetContext;
+            internal IntPtr GetContext;
+            internal IntPtr SetCallbackHandler;
+
+            internal IntPtr SetParam;
+            internal IntPtr GetParam;
+
+            internal IntPtr RegistrationOpen;
+            internal IntPtr RegistrationClose;
+
+            internal IntPtr SecConfigCreate;
+            internal IntPtr SecConfigDelete;
+
+            internal IntPtr SessionOpen;
+            internal IntPtr SessionClose;
+            internal IntPtr SessionShutdown;
+
+            internal IntPtr ListenerOpen;
+            internal IntPtr ListenerClose;
+            internal IntPtr ListenerStart;
+            internal IntPtr ListenerStop;
+
+            internal IntPtr ConnectionOpen;
+            internal IntPtr ConnectionClose;
+            internal IntPtr ConnectionShutdown;
+            internal IntPtr ConnectionStart;
+
+            internal IntPtr StreamOpen;
+            internal IntPtr StreamClose;
+            internal IntPtr StreamStart;
+            internal IntPtr StreamShutdown;
+            internal IntPtr StreamSend;
+            internal IntPtr StreamReceiveComplete;
+            internal IntPtr StreamReceiveSetEnabled;
+        }
+
+        internal delegate uint SetContextDelegate(
+            IntPtr handle,
+            IntPtr context);
+
+        internal delegate IntPtr GetContextDelegate(
+            IntPtr handle);
+
+        internal delegate void SetCallbackHandlerDelegate(
+            IntPtr handle,
+            Delegate del,
+            IntPtr context);
+
+        internal delegate uint SetParamDelegate(
+            IntPtr handle,
+            uint level,
+            uint param,
+            uint bufferLength,
+            byte* buffer);
+
+        internal delegate uint GetParamDelegate(
+            IntPtr handle,
+            uint level,
+            uint param,
+            uint* bufferLength,
+            byte* buffer);
+
+        internal delegate uint RegistrationOpenDelegate(byte[] appName, out IntPtr registrationContext);
+
+        internal delegate void RegistrationCloseDelegate(IntPtr registrationContext);
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct CertHash
+        {
+            internal const int ShaHashLength = 20;
+
+            [MarshalAs(UnmanagedType.ByValArray, SizeConst = ShaHashLength)]
+            internal byte[] ShaHash;
```

Comment:
It would be better to use `fixed` fields for these. `UnmanagedType.ByValArray` is not efficient way to do interop. We prefer to use blittable types where possible for interop (https://github.com/dotnet/runtime/blob/master/docs/coding-guidelines/interop-guidelines.md).

Reply:
These are unused anyways for now. I'll delete them. Are we okay with these in delegates:
```c#
        internal delegate uint SecConfigCreateDelegate(
            IntPtr registrationContext,
            uint flags,
            IntPtr certificate,
            [MarshalAs(UnmanagedType.LPStr)]string principal,
            IntPtr context,
            SecConfigCreateCompleteDelegate completionHandler);```